assign_rights/tests.py

```diff
@@ -1,3 +1,38 @@
-# from django.test import TestCase
+from django.test import TestCase
+from django.urls import reverse
+from rest_framework.test import APIRequestFactory
 
-# Create your tests here.
+from .models import RightsShell
+from .views import RightsAssemblerView
+
+
+class TestModels(TestCase):
+
+    def create_rightsshell(self):
+        return RightsShell.objects.create(
+            rights_id=1,
+            rights_basis="Copyright",
+            copyright_status="copyrighted",
+            determination_date="2020-01-01",
+            note="note",
+            applicable_start_date="2019-02-01",
+            applicable_end_date="2019-03-01",
+            start_date_period=None,
+            end_date_period=None,
+            end_date_open=False,
+            license_terms=None,
+            statute_citation=None
+        )
+
+    def test_rightsshell_creation(self):
+        s = self.create_rightsshell()
+        self.assertTrue(isinstance(s, RightsShell))
+
+
+class TestViews(TestCase):
+
+    def test_rightsassemblerview(self):
+        TestModels.create_rightsshell(self)
+        factory = APIRequestFactory()
+        request = factory.post(reverse('rights-assemble'), {"objects": [{"identifier": 1, "start_date": "2019-12-09", "end_date": "2020-01-01"}, {"identifier": 2, "start_date": "2019-12-01", "end_date": "2020-02-01"}, {"identifier": 3, "start_date": "2019-11-09", "end_date": "2020-03-01"}]}, format='json')
```

Comment:
You're right that there would be an array of identifiers (I'm realizing I implied otherwise @p-galligan, my bad!). However, there would only be an end date, since a start date wouldn't be needed in order to calculate rights statement dates.